pulsar-client-cpp/CMakeLists.txt

```diff
@@ -22,7 +22,11 @@ project (pulsar-cpp)
 
 set(Boost_NO_BOOST_CMAKE ON)
 
-set (CMAKE_CXX_FLAGS "-O3 -g -Wno-deprecated-declarations ${CMAKE_CXX_FLAGS}")
+if (NOT CMAKE_CXX_FLAGS)
```

Comment:
I would rather use a different variable name, since `CMAKE_CXX_FLAGS` is of general use. Something like `CXX_STANDARD` or similar to be use just for this purpose.